src/modules/commander/accelerometer_calibration.cpp

```diff
@@ -613,92 +381,68 @@ calibrate_return do_accel_calibration_measurements(orb_advert_t *mavlink_log_pub
-		fds[i].events = POLLIN;
-	}
-
-	unsigned counts[max_accel_sens] = { 0 };
-	float accel_sum[max_accel_sens][3] {};
+	Vector3f accel_sum[MAX_ACCEL_SENS] {};
+	unsigned counts[MAX_ACCEL_SENS] {};
 
 	unsigned errcount = 0;
-	struct sensor_correction_s sensor_correction; /**< sensor thermal corrections */
 
-	/* try to get latest thermal corrections */
-	if (orb_copy(ORB_ID(sensor_correction), sensor_correction_sub, &sensor_correction) != 0) {
-		/* use default values */
-		memset(&sensor_correction, 0, sizeof(sensor_correction));
+	// sensor thermal corrections
+	uORB::Subscription sensor_correction_sub{ORB_ID(sensor_correction)};
+	sensor_correction_s sensor_correction{};
+	sensor_correction_sub.copy(&sensor_correction);
 
-		for (unsigned i = 0; i < 3; i++) {
-			sensor_correction.accel_scale_0[i] = 1.0f;
-			sensor_correction.accel_scale_1[i] = 1.0f;
-			sensor_correction.accel_scale_2[i] = 1.0f;
-		}
-	}
+	uORB::SubscriptionBlocking<sensor_accel_s> accel_sub[MAX_ACCEL_SENS] {
+		{ORB_ID(sensor_accel), 0, 0},
+		{ORB_ID(sensor_accel), 0, 1},
+		{ORB_ID(sensor_accel), 0, 2},
+	};
 
 	/* use the first sensor to pace the readout, but do per-sensor counts */
 	while (counts[0] < samples_num) {
-		int poll_ret = px4_poll(&fds[0], max_accel_sens, 1000);
-
-		if (poll_ret > 0) {
-
-			for (unsigned s = 0; s < max_accel_sens; s++) {
-				bool changed;
-				orb_check(subs[s], &changed);
-
-				if (changed) {
-
-					sensor_accel_s arp;
-					orb_copy(ORB_ID(sensor_accel), subs[s], &arp);
-
-					// Apply thermal offset corrections in sensor/board frame
-					if (s == 0) {
-						accel_sum[s][0] += (arp.x - sensor_correction.accel_offset_0[0]);
-						accel_sum[s][1] += (arp.y - sensor_correction.accel_offset_0[1]);
-						accel_sum[s][2] += (arp.z - sensor_correction.accel_offset_0[2]);
-
-					} else if (s == 1) {
-						accel_sum[s][0] += (arp.x - sensor_correction.accel_offset_1[0]);
-						accel_sum[s][1] += (arp.y - sensor_correction.accel_offset_1[1]);
-						accel_sum[s][2] += (arp.z - sensor_correction.accel_offset_1[2]);
-
-					} else if (s == 2) {
-						accel_sum[s][0] += (arp.x - sensor_correction.accel_offset_2[0]);
-						accel_sum[s][1] += (arp.y - sensor_correction.accel_offset_2[1]);
-						accel_sum[s][2] += (arp.z - sensor_correction.accel_offset_2[2]);
-
-					} else {
-						accel_sum[s][0] += arp.x;
-						accel_sum[s][1] += arp.y;
-						accel_sum[s][2] += arp.z;
+		if (accel_sub[0].updatedBlocking(100000)) {
+			for (unsigned s = 0; s < MAX_ACCEL_SENS; s++) {
+				sensor_accel_s arp;
+
+				if (accel_sub[s].update(&arp)) {
+					// fetch optional thermal offset corrections in sensor/board frame
+					Vector3f offset{0, 0, 0};
+					sensor_correction_sub.update(&sensor_correction);
+
+					if (sensor_correction.timestamp > 0) {
+						for (uint8_t i = 0; i < MAX_ACCEL_SENS; i++) {
+							if (sensor_correction.accel_device_ids[i] == arp.device_id) {
+								switch (sensor_correction.accel_device_ids[i]) {
```

Comment:
Is the accel device id `0`, `1`, or `2`?

Or should this be?
```suggestion
								switch (i) {
```

Reply:
Good catch. This is supposed to be finding and copying the relevant offsets for a particular device id.